rust/mlt-nom/src/decoder/integer.rs

```diff
@@ -0,0 +1,18 @@
+use crate::v01::{Stream, StreamMeta};
+use crate::{MltError, MltRefResult};
+
+pub fn decode_int_stream(
+    input: &[u8],
+    metadata: &StreamMeta,
+    is_signed: bool,
+) -> MltRefResult<'_, Vec<i32>> {
```

Comment:
there is no point to use `MltRefResult` here -- you are not returning a reference to the original data - that's what Stream creation code does. Your code here converts `Stream` instances into owned `Vec<T>`.  So just use `Result<Vec<i32>, MltError>`. Same elsewhere